views: moderation: Add unit tests, proper response instead of "Yo", apiDoc clarifications

fliiiix opened the pull request:
Things this PR addresses:
* Replace Yo response from moderation with a more appropriate response
* Add tests for moderation
* Add tests for unsubscribe and fix doc string

Contributes to #755

ix5 commented on isso/tests/test_comments.py:
```diff
@@ -604,6 +604,115 @@ def testAddComment(self):
+
+        # GET should return some html form
+        action = "activate"
+        rv_activat_get = self.client.get('/id/%d/%s/%s' % (id_, action, signed))
```
`rv_activate_get`

ix5 commented on isso/tests/test_comments.py:
```diff
@@ -604,6 +604,115 @@ def testAddComment(self):
+        self.assertIn(b"Activate: Are you sure?", rv_activat_get.data)
+        self.assertIn(b"http://invalid.localmoderated#isso-1", rv_activat_get.data)
+
+        # GET should return some html form
```
Any reason for duplicating this test?

fliiiix replied:
nope this is a rebase mistake i overlooked

ix5 commented on isso/tests/test_comments.py:
```diff
@@ -604,6 +604,115 @@ def testAddComment(self):
+
+        # Create new comment, should have mode=2 (pending moderation)
+        rv = self.client.post(
+            '/new?uri=moderated', data=json.dumps({"text": "..."}))
```
Maybe use `uri=/moderated` (with slash) so that later on, the assert with `invalid.local/moderated` instead of `invalid.localmoderated` is not as confusing

ix5 commented on isso/tests/test_comments.py:
```diff
@@ -604,6 +604,115 @@ def testAddComment(self):
+        self.assertEqual(json.loads(rv_edit.data)["id"], id_)
+        self.assertEqual(self.app.db.comments.get(id_)["text"], "new text")
+
+        # Wrong action on comment is handles by the routing
```
handled

ix5 commented on isso/tests/test_comments.py:
```diff
@@ -604,6 +604,115 @@ def testAddComment(self):
 
+    def testModerateComment(self):
+
+        id_ = 1
```
Using `id_` instead of my usage of private `_id` is much smarter, good catch!

fliiiix replied:
its also what PEP8 tells you to do ;)

ix5 commented on isso/tests/test_comments.py:
```diff
@@ -604,6 +604,115 @@ def testAddComment(self):
+        self.assertEqual(rv_unsubscribe_get.status_code, 200)
+        self.assertIn(b"Successfully unsubscribed", rv_unsubscribe_get.data)
+
+        # Incomplet key should fail
```
Incomplete, same for `rv_incomplete_key`

ix5 commented on isso/views/comments.py:
```diff
@@ -649,7 +646,7 @@ def unsubscribe(self, environ, request, id, email, key):
 
     @apiSuccessExample Using POST:
-        Yo
+        Comment is deleted
```
"Comment deleted" / "Comment has been deleted" ?

ix5 commented on isso/views/comments.py:
```diff
@@ -689,7 +686,7 @@ def moderate(self, environ, request, id, action, key):
                 self.comments.activate(id)
             self.signal("comments.activate", thread, item)
-            return Response("Yo", 200)
+            return Response("Comment is activated", 200)
```
"Comment activated" / "Comment has been activated" ?

ix5 commented on isso/views/comments.py:
```diff
@@ -704,7 +701,7 @@ def moderate(self, environ, request, id, action, key):
                 'hash', (item['email'] or item['remote_addr']).encode('utf-8'))
             self.signal("comments.delete", id)
-            return Response("Yo", 200)
+            return Response("Comment is deleted", 200)
```
dito

ix5 reviewed (changes requested):
Nice effort! Some minor nits here and there.

Please also keep commit message titles under 50 chars (or at least under 70, if otherwise not possible), else GitHub will hard-break the title with ellipses.

fliiiix commented:
Thanks for the review :tada:  i addressed all your feedback, take an other look if you spot any issues

Except the thing with the git commit message length, im not to sure what i should do there I can drop the prefix to make them shorter, but i guess the idea is to use prefixes? I can create super short unspecific commit messages which i feel would be worse. 
So i am a bit lost how i should solve that any ideas?

ix5 commented on isso/views/comments.py:
```diff
@@ -561,25 +561,19 @@ def delete(self, environ, request, id, key=None):
         The email address of the subscriber.
     @apiParam {string} key
-        The key to authenticate the subscriber.
+        The key to authenticate the subscriber. Containing a list with the keyword unsubscribe and the email ['unsubscribe', email].
```
Maybe mention here that the key is also base64-encoded.
The stray `['unsubscribe', email]` here is also a bit confusing.
Maybe put it on a new line with "Expected syntax" or something.

ix5 replied:
Pushed to your branch with my suggested change. I've left the base64 encoding part out because there's also a signature attached, which would just be confusing.

fliiiix replied:
thanks :+1:

ix5 reviewed (commented):
> Except the thing with the git commit message length, im not to sure what i should do there I can drop the prefix to make them shorter, but i guess the idea is to use prefixes? I can create super short unspecific commit messages which i feel would be worse. So i am a bit lost how i should solve that any ideas?

Don't worry too much about that. Those commit message conventions are just that, _conventions_. I personally use abbreviations or more terse wording and kind of "force" myself to keep below 50 chars, as I've set my editor to "warn" me when I go over.

E.g.
```
views: comments: moderation: Replace Yo response with a more descriptive text
```
I'd replace with
```
views: comments: Proper response instead of 'Yo'
```

But again, that's just stylistic pet peeves. You'll notice the usefulness of short commit messages when browsing git logs and being annoyed at the cut-off messages ending in ellipses.

---

Thanks for those tests. One inline comment, everything else looks marvelous!

ix5 commented:
Verified using `make test` and `npm i apidoc; npx apidoc --config apidoc.json --input isso/ --output _apidoc/`

Thank you for your contribution!
